pyvista/utilities/features.py

```diff
@@ -56,6 +56,15 @@ def voxelize(mesh, density=None, check_surface=True):
     if isinstance(density, (list, set, tuple)):
         density_x, density_y, density_z = density
 
+    # check and pre-process input mesh
+    surface = mesh.extract_geometry()  # filter preserves topology
+    if not surface.faces.size:
+        # we have a point cloud or an empty mesh
+        raise ValueError('Input mesh must have faces for voxelization.')
```

Comment:
And now we explicitly check for faces. We could still have isolated points along with a manifold mesh, or multiple disjoint manifold meshes (which might or might not work reliably), but checking for that is probably not worth it in the majority of use cases.

Reply:
Thanks. This works and allows for a quick fail for a good reason.